### PR TITLE
🚚 Move `/reader/subscribe` to `/reader/follow`

### PR DESCRIPTION
**Pull request by williamchong007**

### CODE REVIEW

File: src/util/api/index.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ export const getPaypalPaymentPageURL = (likerId, custom) => {
 };
 export const getFetchLikedUserApi = () => `/api/reader/index`;
 export const getFetchUserArticlesAPI = user => `/api/reader/user/${user}`;
-export const getSubscribeUserAPI = user => `/api/reader/subscribe/user/${user}`;
+export const getSubscribeUserAPI = user => `/api/reader/follow/user/${user}`;
```
Comment (nwingt) [inline src/util/api/index.js]: getFollowUserAPI?

**Comment by nwingt** [comment]
Discussion, I think `/reader/user/` is ambiguous too, prefer `/reader/author`

File: src/store/modules/actions/reader.js [hunk 2]
```diff
@@ -2,24 +2,24 @@ import * as types from '@/store/mutation-types';
 import * as api from '@/util/api';
 
 export async function fetchReaderIndex({ commit }) {
-  const { list, unsubscribedUsers } = await this.$axios.$get(
+  const { list, unfollowdUsers } = await this.$axios.$get(
```
Comment (nwingt) [inline src/store/modules/actions/reader.js]: unfollow**e**dUsers

File: src/store/modules/actions/reader.js [hunk 3]
```diff
@@ -2,24 +2,24 @@ import * as types from '@/store/mutation-types';
...
     api.getFetchLikedUserApi()
   );
-  commit(types.READER_SET_USER_LIST, list);
-  commit(types.READER_SET_UNSUB_USER_LIST, unsubscribedUsers);
+  commit(types.READER_SET_FOLLOW_USER_LIST, list);
+  commit(types.READER_SET_UNFOLLOW_USER_LIST, unfollowdUsers);
```
Comment (nwingt) [inline src/store/modules/actions/reader.js]: unfollow**e**dUsers

File: src/server/api/routes/reader/index.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    }
+    const [{ data }, userDoc] = await Promise.all([
+      apiFetchLikedUser(req),
+      userCollection.doc(req.session.user).get(),
+    ]);
+    const userSet = new Set(data.list);
+    const { followedUsers = [], unfollowedUsers = [] } = userDoc.data();
+    followedUsers.forEach(u => userSet.add(u));
+    unfollowedUsers.forEach(u => userSet.delete(u));
+    res.json({ list: Array.from(userSet), unfollowedUsers });
```
Comment (nwingt) [inline src/server/api/routes/reader/index.js]: Discussion: Is it necessary to return unfollowedUsers?

Comment (williamchong007) [reply src/server/api/routes/reader/index.js]: yes, if you want to un-unfollow

**Pull request merged**